unstructured/ingest/v2/processes/connectors/opensearch.py

```diff
@@ -0,0 +1,124 @@
+import json
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.utils.data_prep import generator_batching_wbytes
+from unstructured.ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+)
+from unstructured.ingest.v2.logger import logger
+from unstructured.ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+    add_destination_entry,
+)
+from unstructured.ingest.v2.processes.connectors.elasticsearch import (
+    ElasticsearchUploaderConfig,
+    ElasticsearchUploadStager,
+    ElasticsearchUploadStagerConfig,
+)
+from unstructured.utils import requires_dependencies
+
+if TYPE_CHECKING:
+    from opensearchpy import OpenSearch
+
+CONNECTOR_TYPE = "opensearch"
+
+"""Since the actual OpenSearch project is a fork of Elasticsearch, we are relying
+heavily on the Elasticsearch connector code, inheriting the functionality as much as possible."""
+
+
+@dataclass
+class OpenSearchAccessConfig(AccessConfig):
+    password: Optional[str] = enhanced_field(default=None, sensitive=True)
+    use_ssl: bool = False
+    verify_certs: bool = False
+    ssl_show_warn: bool = False
+    ca_certs: Optional[str] = None
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+
+
+@dataclass
+class OpenSearchConnectionConfig(ConnectionConfig):
+    hosts: Optional[list[str]] = None
+    username: Optional[str] = None
+    access_config: OpenSearchAccessConfig = enhanced_field(sensitive=True)
+
+    def to_dict(self, **kwargs) -> dict[str, json]:
+        d = super().to_dict(**kwargs)
+        d["use_ssl"] = self.access_config.use_ssl
```

Comment:
@rbiseck3 is there a more elegant way to "unpack" the `access_config` since its a nested dictionary at this level?

Reply:
I guess I can just update the dictionary with a dict comprehension.

Reply:
Got ya. I'll stick with what I have. That one looks less fun.